File: TaskMasterAPI/models/attached_file.py
from database import db


class AttachedFile(db.Model):
    task_id = db.Column("taskId", db.Integer, db.ForeignKey('task.id'), primary_key=True)
    url = db.Column(db.String(255), primary_key=True)
    file_name = db.Column("fileName", db.String(255))

File: task_master/lib/domain/model/attached_file.dart
class AttachedFile {
  int taskId;
  String url;
  String fileName;

  AttachedFile({required this.taskId, required this.url, required this.fileName});

  factory AttachedFile.fromJson(Map<String, dynamic> json) {
    return AttachedFile(
      taskId: json['task_id'],
      url: json['url'],
      fileName: json['fileName'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'task_id': taskId,
      'url': url,
      'fileName': fileName,
    };
  }
}
